### DISCUSSION TITLE
Can do Multiple Digital signing for a MSI using wixsharp?

### DISCUSSION THREAD

**Discussion by Naganathan** [oleg-shilo/wixsharp#1628] [Q&A] [answered]
I am using Wixsharp code to generate setup installation for my project. 
I need to do Multiple digital Signature signing for a MSI before pack to generate setup/bootstrap using WixSharp.
Is there options do that?

**Comment by oleg-shilo** [comment]
Look at the "Signing" sample in the samples folder.
Do the search in the GitHub project and it will show you both matches - wiki and the code:
 
 
![image](https://github.com/user-attachments/assets/60a92ad3-58c8-4301-a16d-6d86989cefe9)

**Reply by Naganathan** [reply]
Hi Olego,

That signing sample contains only one signing to the setup but my question/expectation needs to do one or more signing to the setup/MSI.

![image](https://github.com/user-attachments/assets/78c39ad9-1663-4c1b-8ff5-0fb4251b6e18)

Thanks,

**Reply by oleg-shilo** [reply]
I am not sure we are talking about the same things. 
The code sample shows how to sign all binary files (multiple files) in the msi. Though if you meant some other scenario then the sample is not relevant for you.

What exactly do you mean by "Multiple Digital Signing"?

**Reply by Naganathan** [reply]
We needs to do 2 sign for a MSI. like 

`
/// 1st Signature
DigitalSignature = new DigitalSignature
{
PfxFilePath = "wixsharp.pfx",
Password = "my_password1",
Description = "MyProduct",
HashAlgorithm = HashAlgorithmType.sha256,
TimeUrl = new Uri("http://timestamp.verisign.com/scripts/timstamp.dll"),
OptionalArguments = $"/a /v /s My /sha1 {Certificate1} /fd sha256"
}

`
` 
/// 2nd Signature
DigitalSignature = new DigitalSignature
{
PfxFilePath = "wixsharp.pfx",
Password = "my_password2",
Description = "MyProduct",
HashAlgorithm = HashAlgorithmType.sha256,
TimeUrl = new Uri("http://timestamp.verisign.com/scripts/timstamp.dll"),
OptionalArguments = $"/a /v /s My /sha1 {Certificate2} /fd sha256"
}

`

I understood there is only one DigitalSignature property for ManagedProject. Can we achieve this any other ways?

**Reply by oleg-shilo** [reply, marked as answer]
Of course. The `Project.DigitalSignature` is really a very simple feature that doesn't do much more than just secure your sign tool.

You still have the all flexibility of signing after the build:

```C#
var msiFile = project.BuildMsi();

Process.Start("signtool.exe", $"{maliFile}.... ").WaitForExit();

```
